### ISSUE TITLE
CSVDecoder: No Long and Int out of range exceptions

### ISSUE THREAD

**Issue by Pavel38l** [FasterXML/jackson-dataformats-text#485]
I have a problem when I try to process a CSV file entered by the user.
If the numeric values ​​are outside the limits of Int or Long, then there is no way I can notify the user about incorrect values ​​that are too large.

Values ​​are corrupted by [`_getBigInteger().intValue()`](https://github.com/FasterXML/jackson-dataformats-text/blob/c94cb960184c5b9c4b3260ade0f54c5d115e88eb/csv/src/main/java/com/fasterxml/jackson/dataformat/csv/impl/CsvDecoder.java#L1376) and [`_getBigInteger().longValue()`](https://github.com/FasterXML/jackson-dataformats-text/blob/c94cb960184c5b9c4b3260ade0f54c5d115e88eb/csv/src/main/java/com/fasterxml/jackson/dataformat/csv/impl/CsvDecoder.java#L1403) in CsvDecoder
And no error occurs
Is it possible to use the methods `_getBigInteger().intValueExact()` and `_getBigInteger().longValueExact()` which check boundaries and cause an error?

Example (Kotlin):
CSV file:
```
testInt,testLong
111111111111111111111111111111111111111111,2222222222222222222222222222222222222222
```
Entity:
```
import com.fasterxml.jackson.annotation.JsonProperty

open class TestEntity(
        @JsonProperty("testInt")  val testInt: Int,
        @JsonProperty("testLong") val testLong: Long,
)
```
```
import com.fasterxml.jackson.dataformat.csv.CsvMapper
import com.fasterxml.jackson.dataformat.csv.CsvParser
import java.io.File
import java.io.InputStream

class TestClass {
    fun readConvertFile() {
        val file = File("example.csv")
        val entities = readCsv(TestEntity::class.java, file.inputStream())
        println(entities[0].testInt)
        println(entities[0].testLong)
    }

    companion object {
        private val csvMapper = CsvMapper()
            .enable(CsvParser.Feature.SKIP_EMPTY_LINES)
            .enable(CsvParser.Feature.TRIM_SPACES)
            .enable(CsvParser.Feature.WRAP_AS_ARRAY)
            .enable(CsvParser.Feature.INSERT_NULLS_FOR_MISSING_COLUMNS)

        fun <T> readCsv(classArg: Class<T>, stream: InputStream): List<T> {
            val schema = csvMapper.schemaFor(classArg).withHeader().withColumnReordering(true).withNullValue("")
            val reader = csvMapper.readerFor(classArg).with(schema)
            return reader.readValues<T>(stream).readAll()
        }
    }
}
```
ER: some error message like jackson ObjectMapper `Numeric value (111111111111111111111111111111111111111111) out of range of int`
AR: Values corrupted
```
-954437177
-1121560256615750770
```

I use
```
<dependency>
    <groupId>com.fasterxml.jackson.dataformat</groupId>
    <artifactId>jackson-dataformat-csv</artifactId>
    <version>2.17.2</version>
</dependency>
```

**Comment by cowtowncoder** [comment]
Quick question: which Jackson version is this with? If not 2.17(.2), then would be good to verify issue with 2.17.2.
If 2.17.2, there is `2.18.0-SNAPSHOT` -- there have been changes to number conversions.

Aside from that: just to make sure -- you would like to get an appropriate exception for value overflow wrt `int`/`Integer` and `long`/`Long`? If so, makes sense.

**Comment by Pavel38l** [comment]
@cowtowncoder hi
I used
```
<dependency>
    <groupId>com.fasterxml.jackson.core</groupId>
    <artifactId>jackson-databind</artifactId>
    <version>2.17.2</version>
</dependency>
```
and jackson-dataformat-csv: 2.17.2
Version 2.18.0 is not yet available in the central Maven repository, I don't know how to check it
In addition, the CSVDecoder class branch 2.18 contains the same methods that I listed above
https://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-databind


Yes, that's right, I would like to receive an error about the overflow of Int and Long values

**Comment by cowtowncoder** [comment]
2.18.0 would be `2.18.0-SNAPSHOT` (having Sonatype OSS Repo configured). But as you say, it is unlikely to differ in behavior.

**Comment by cowtowncoder** [comment]
Ok, was able to reproduce the issue; adding failing test via #487.

**Comment by cowtowncoder** [comment]
Fixed, to be include in 2.18.0.

Oddly enough, coercion limits were enforced if converting from floating-point values.
And the reason this was missing was because it's not `CsvParser` that tracks number values, conversions, but `CsvDecoder` -- which cannot rely on `ParserBase` that has necessary checks. So code is copy-pasted.
